File: Homework 2/Codes/Gozubuyuk_Bulut.py
print("------------------pretest")
# pre test
i = 0
while i < 3:
    print(i)
    i += 1

print("------------------posttest")

# post test

i = 0
while True:
    print(i)
    i += 1
    if not i < 3:
        break

print("------------------break")

# break

i = 0
while i < 3:
    print(i)
    if i == 1:
        break
    i += 1

print("------------------continue")

# continue

i = 0
while i < 3:
    i += 1
    if i == 2:
        continue
    print(i)

File: Homework 2/Codes/Gozubuyuk_Bulut.dart
void main() {
  print('-----------------pretest');
  // pre test
  var i = 0;
  while (i < 5) {
    print(i);
    i++;
  }

  print('-----------------posttest');
  // post test
  i = 0;
  do {
    print(i);
    i++;
  } while (i < 5);

  print('-----------------break');
  // unlabeled break
  i = 0;
  while (i < 5) {
    i++;
    print(i);
    break;
  }

  print('-----------------continue');
  // continue
  i = 0;
  while (i < 5) {
    i++;
    if (i == 2) { continue; }
    print(i);
  }

  print('-----------------labeledbreak');
  // labeled break
  i = 0;
  label:
  while (true) {
    while (true) {
      print(i);
      if (i == 1) break label;
      i++;
    }
  }

  print('-----------------labeledcontinue');
  // labeled continue
  i = 0;
  var j = 0;
  anotherlabel:
  while (i < 2) {
    while (j < 2) {
      print("$i,$j");
      j++;
      if (i == j) continue anotherlabel;
    }
    i++;
    print(i);
  }

}